### ISSUE TITLE
Intersect misses dimensions when stacked

### ISSUE THREAD

**Issue by mpu-creare** [creare-com/podpac#114]
**Description**
When intersecting a grid using a point coordinate, the full grid is returned because the dimensions are miss-matched

**Steps to Reproduce**
```
In [1]: from podpac.core.coordinates import clinspace, Coordinates

In [2]: c_grid = Coordinates([clinspace(0, 1, 10), clinspace(0, 1, 11)], dims=['lat', 'lon'])

In [3]: c_pts = Coordinates([[0.5, 0.5]], dims=['lat_lon'])

In [4]: c_grid.intersect(c_pts)
Out[4]:
Coordinates
        lat: UniformCoordinates1d(lat): Bounds[0.0, 1.0], N[10], ctype['midpoint']
        lon: UniformCoordinates1d(lon): Bounds[0.0, 1.0], N[11], ctype['midpoint']

In [5]: c_grid.intersect(c_pts, outer=True)
Out[5]:
Coordinates
        lat: UniformCoordinates1d(lat): Bounds[0.0, 1.0], N[10], ctype['midpoint']
        lon: UniformCoordinates1d(lon): Bounds[0.0, 1.0], N[11], ctype['midpoint']

```

**Expected Behavior**
I would expect behavior similar to the following:
```
In [6]: c_grid1pt = Coordinates([0.5, 0.5], dims=['lat', 'lon'])
In [7]: c_grid.intersect(c_grid1pt)
Out[7]:
Coordinates
        lat: ArrayCoordinates1d(lat): Bounds[nan, nan], N[0], ctype['midpoint']
        lon: UniformCoordinates1d(lon): Bounds[0.5, 0.5], N[1], ctype['midpoint']

In [8]: c_grid.intersect(c_grid1pt, outer=True)
Out[8]:
Coordinates
        lat: UniformCoordinates1d(lat): Bounds[0.4444444444444444, 0.5555555555555556], N[2], ctype['midpoint']
        lon: UniformCoordinates1d(lon): Bounds[0.4, 0.6000000000000001], N[3], ctype['midpoint']
````

**Additional Notes**
Problem happens here: https://github.com/creare-com/podpac/blob/develop/podpac/core/coordinates/coordinates1d.py#L236 

```
self.name == 'lat'
self.name != 'lat_lon'
other.dims == 'lat_lon'

**Comment by mpu-creare** [comment]
@jmilloy  FYI. I have a fix on the SMAP branch.

**Comment by mpu-creare** [comment]
Fixed here 6fa21dd2ebe9258d8337fc828baf416f7b26c938
